website/docs/internals/provider-network-mirror-protocol.html.md

```diff
@@ -0,0 +1,278 @@
+
+The URLs are shown with the convention that a path portion with a colon `:`
+prefix is a placeholder for a dynamically-selected value, while all other
+path portions are literal. For example, in `:hostname/:namespace/:type/index.json`,
+the first three path portions are placeholders while the third is literally
+the string "index.json".
+
+The example requests in the following sections will assume the example mirror
+base URL from the above CLI configuration example.
+
+### Authentication
+
+If the CLI configuration includes
+[credentials](/docs/commands/cli-config.html#credentials) for the hostname
+given in the network mirror base URL, Terraform will include those credentials
+in its requests for operations described below.
+
+If the given URL uses a non-standard port number (other than 443) then the
+credentials must be associated with a hostname that includes the port number,
+such as `terraform.example.com:8443`.
+
+Terraform does _not_ send credentials when retrieving the archives whose
+URLs are given in the "List Available Installation Packages" response below.
+If a particular mirror considers the distribution packages themselves to be
+sensitive then it must use cryptographically-secure, user-specific, and
+time-limited URLs in the metadata response. Strategies for doing so are out
+of scope of this protocol documentation.
+
+## List Available Versions
+
+This operation determines which versions are currently available for a
+particular provider.
+
+| Method | Path                                    | Produces           |
+|--------|-----------------------------------------|--------------------|
+| `GET`  | `:hostname/:namespace/:type/index.json` | `application/json` |
+
+### Parameters
+
+* `hostname` (required): the hostname portion of the address of the requested
+  provider.
+* `namespace` (required): the namespace portion of the address of the requested
+  provider.
+* `type` (required): the type portion of the address of the requested provider.
+
+### Sample Request
+
+```
+curl 'https://terraform.example.com/providers/registry.terraform.io/hashicorp/random/index.json'
+```
+
+### Sample Response
+
+```json
+{
+  "versions": {
+    "2.0.0": {},
+    "2.0.1": {}
+  }
+}
+```
+
+### Response Properties
+
+A successful result is a JSON object containing a single property `versions`,
+which must be a JSON object.
+
+Each of the property names of the `versions` object represents an available
+version number. The property values must be objects, but no properties are
+currently defined for those objects. Future versions of this protocol may
+define optional per-version properties for Terraform to use as installation
+hints, so implementations of the current version should leave those objects
+empty.
+
+Return `404 Not Found` to signal that the mirror does not have a provider
+with the given address.
+
+## List Available Installation Packages
+
+This operation returns download URLs of and associated metadata about the
```

Comment:
```suggestion
This operation returns download URLs and associated metadata about the
```

I think it's ok to leave off the 'of' here but it's more of a style suggestion, I will not argue if you ignore this one!

Reply:
"download URLs about" feels weird to me, but both "download URLs for" and "associated metadata for" both seem to work, so I think I'm going to address this by removing the "of" and changing "about" to "for", so the new preposition will be shared between both of the noun phrases. Thanks!

(There's a similar passage to this in the Provider Registry Protocol page, which I used as the base here. However, to keep this already-rather-large PR focused I'm going to ignore that for now and focus on editing this new docs page.)